controller/appcontroller.go

```diff
@@ -1436,35 +1436,22 @@ func (ctrl *ApplicationController) processRequestedAppOperation(app *appv1.Appli
 	}
 	ts.AddCheckpoint("initial_operation_stage_ms")
 
-	// Call GetDestinationCluster to validate the destination cluster.
-	if _, err := argo.GetDestinationCluster(context.Background(), app.Spec.Destination, ctrl.db); err != nil {
-		state.Phase = synccommon.OperationFailed
-		state.Message = err.Error()
-	} else {
-		ctrl.appStateManager.SyncAppState(app, state)
-	}
```

Comment:
I'm guessing you're making this change because we perform the same check in SyncAppState anyway. Is that the motivation for dropping the check?

Reply:
yes. if syncAppState/compareAppState require a valid destination, they will do the check. And they are already doing it because they do need a valid destination.